src/test/java/hudson/plugins/mercurial/MercurialSCMSource2Test.java

```diff
@@ -61,7 +61,7 @@
     @ClassRule public static DockerClassRule<MercurialContainer> docker = new DockerClassRule<>(MercurialContainer.class);
     @Rule public TemporaryFolder tmp = new TemporaryFolder();
 
-    @Issue({"JENKINS-42278", "JENKINS-48867"})
+    @Issue({"JENKINS-42278", "JENKINS-46851", "JENKINS-48867"})
     @Test public void withCredentialsId() throws Exception {
```

Comment:
Skipped? https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fmercurial-plugin/detail/PR-151/2/tests/

Reply:
I am afraid so—tests of the branch source require a `hg` installation on the _controller_ side, which `JenkinsRule` + `docker-fixtures` does not support. I run these tests when doing a local release.